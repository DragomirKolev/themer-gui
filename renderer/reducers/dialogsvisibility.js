import {
  TIPS_DIALOG_OPEN,
  EXPORT_DIALOG_OPEN,
  EXPORT_PROGRESS_DIALOG_OPEN,
  CLOSE_DIALOGS,
  PREFILL_DIALOG_OPEN,
  HELP_DIALOG_OPEN,
} from '../actions';

const defaultDialogsVisibility = {
  export: false,
  tips: false,
  exportProgress: false,
  help: false,
};

export const dialogsVisibilityReducer = (state = defaultDialogsVisibility, action) => {
  switch (action.type) {
    case EXPORT_DIALOG_OPEN:
      return {
        ...state,
        export: true,
      };
    case TIPS_DIALOG_OPEN:
      return {
        ...state,
        tips: true,
      };
    case EXPORT_PROGRESS_DIALOG_OPEN:
      return {
        ...state,
        exportProgress: true,
      };
    case PREFILL_DIALOG_OPEN:
      return {
        ...state,
        prefill: true,
      };
    case HELP_DIALOG_OPEN:
      return {
        ...state,
        help: true,
      };
    case CLOSE_DIALOGS:
      return Object.keys(state).reduce((reduced, key) => ({ ...reduced, [key]: false }), {});
    default:
      return state;
  }
};
