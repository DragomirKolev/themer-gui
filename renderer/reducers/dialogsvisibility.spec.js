import { dialogsVisibilityReducer } from './dialogsvisibility';
import {
  helpDialogOpen,
  exportDialogOpen,
  exportProgressDialogOpen,
  closeDialogs,
} from '../actions';

describe('dialogsVisibilityReducer', () => {
  it('should initialize to all dialogs closed', () => {
    const defaultState = dialogsVisibilityReducer(undefined, {});
    expect(Object.values(defaultState).some(Boolean)).toBe(false);
  });
  it('should open the help dialog', () => {
    const newState = dialogsVisibilityReducer(undefined, helpDialogOpen());
    expect(newState.help).toBe(true);
  });
  it('should open the export dialog', () => {
    const newState = dialogsVisibilityReducer(undefined, exportDialogOpen());
    expect(newState.export).toBe(true);
  });
  it('should open the export progress dialog', () => {
    const newState = dialogsVisibilityReducer(undefined, exportProgressDialogOpen());
    expect(newState.exportProgress).toBe(true);
  });
  it('should close all dialogs', () => {
    const initialState = {
      export: true,
      help: false,
      exportProgress: false,
    };
    const expected = {
      ...initialState,
      export: false,
    };
    expect(dialogsVisibilityReducer(initialState, closeDialogs())).toEqual(expected);
  });
});
