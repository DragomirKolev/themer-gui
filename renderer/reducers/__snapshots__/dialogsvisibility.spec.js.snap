// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialogsVisibilityReducer should close all dialogs 1`] = `
Object {
  "export": false,
  "exportProgress": false,
  "help": false,
  "tips": false,
}
`;

exports[`dialogsVisibilityReducer should initialize to all dialogs closed 1`] = `
Object {
  "export": false,
  "exportProgress": false,
  "help": false,
  "tips": false,
}
`;

exports[`dialogsVisibilityReducer should open the export dialog 1`] = `
Object {
  "export": true,
  "exportProgress": false,
  "help": false,
  "tips": false,
}
`;

exports[`dialogsVisibilityReducer should open the export progress dialog 1`] = `
Object {
  "export": false,
  "exportProgress": true,
  "help": false,
  "tips": false,
}
`;

exports[`dialogsVisibilityReducer should open the help dialog 1`] = `
Object {
  "export": false,
  "exportProgress": false,
  "help": true,
  "tips": false,
}
`;

exports[`dialogsVisibilityReducer should open the prefill dialog 1`] = `
Object {
  "export": false,
  "exportProgress": false,
  "help": false,
  "prefill": true,
  "tips": false,
}
`;

exports[`dialogsVisibilityReducer should open the tips dialog 1`] = `
Object {
  "export": false,
  "exportProgress": false,
  "help": false,
  "tips": true,
}
`;
